### ISSUE TITLE
How about performance in comparison with paper?

### ISSUE THREAD

**Issue by John1231983** [NanqingD/DeepLabV3-Tensorflow#3]
Great to know someone implement Deeplabv3. It is not an issue, I just want to know the performance what you have achieved. Could you tell me how about mIoU did you achieve? Thanks so much.

This is my log, but it is too worse. One more thing, your code did not provide resnet pretrain and did not use it, I guess the reason of below log is training from scratch

```
step 8902, tot_loss = 1.370319, seg_loss = 1.045083, reg_loss = 0.325237, mean_iou: 0.031608, lr: 0.009866(0.552 sec/step)
step 8903, tot_loss = 1.761859, seg_loss = 1.436624, reg_loss = 0.325235, mean_iou: 0.031609, lr: 0.009866(0.535 sec/step)

```

**Comment by NanqingD** [comment]
Hi, John. Thanks for your interest in this repo.

1) Performance
![mean_iou](https://user-images.githubusercontent.com/16928463/34000301-dce6835c-e0ba-11e7-99a3-fa4de41a979d.png)
The convergence of this implementation is slow compared to the paper. Actually, I am still running it.
There are few possibilities:
1. Initialization
Yes, you are correct, this is training from scratch. The paper didn't specify how to initialize for training from scratch, so I take some liberty here.
2. Pretraining
With ResNet pre-trained weights, the training should be much faster, as the paper did. I will implement it later. One thing should also be noticed that the paper tried the pre-training on MS COCO, this is something should be done in the future.
3. Hardware
As the paper said:
    > Our baseline employs simply one replica and requires training time 3.65 days with a K80 GPU. As shown in Tab. 11, we found that the performance of using multiple replicas does not drop compared
to the baseline. However, training time with 32 replicas is significantly reduced to 2.74 hours.
Currently, the training is on single 1080 Ti. Even though I still doubt that they finished the baseline in 60K iterations, it wouldn't take 3.5+ days.
4. Parallel
One solution is to use multi-GPU in training like the classic example in Tensorflow. I am implementing that.

This is a really new repo I did very recently and is still under development, let me know if you find more problems. Thank you very much.

**Comment by NanqingD** [comment]
@John1231983 

I just play around with the learning rate, it seems that it is quite sensitive to learning rate.

Here is what I got:
```
step 417382, tot_loss = 0.292986, seg_loss = 0.100410, reg_loss = 0.192576, mean_iou: 0.753419, lr: 0.005000(0.596 sec/step)
step 417383, tot_loss = 0.413173, seg_loss = 0.220599, reg_loss = 0.192573, mean_iou: 0.753504, lr: 0.005000(0.608 sec/step)
step 417384, tot_loss = 0.313302, seg_loss = 0.120731, reg_loss = 0.192571, mean_iou: 0.753615, lr: 0.005000(0.607 sec/step)
step 417385, tot_loss = 0.384654, seg_loss = 0.192086, reg_loss = 0.192569, mean_iou: 0.753578, lr: 0.005000(0.606 sec/step)
step 417386, tot_loss = 0.468697, seg_loss = 0.276131, reg_loss = 0.192566, mean_iou: 0.753528, lr: 0.005000(0.598 sec/step)
step 417387, tot_loss = 0.270273, seg_loss = 0.077709, reg_loss = 0.192563, mean_iou: 0.753429, lr: 0.005000(0.596 sec/step)
step 417388, tot_loss = 0.468347, seg_loss = 0.275786, reg_loss = 0.192561, mean_iou: 0.753611, lr: 0.005000(0.600 sec/step)
step 417389, tot_loss = 0.284128, seg_loss = 0.091569, reg_loss = 0.192559, mean_iou: 0.753490, lr: 0.005000(0.598 sec/step)
```

**Comment by John1231983** [comment]
Thanks for your information. It is very useful for me. The mIoU comes from train+val set, I remember that it is about 86% in deeplabv2 when we use train+val as training set.
I also think using resnet pretrain may have challenge because the deeplabv3's encoder using modification of resnet
One more thing,  you used the trainval.txt to generate tfrecord for training. It means that you are using both training and validation set for generating tfrecord. It may not good for checking hyperparameter. Normally, we often use training set for training and then evaluate in the validation set to select best hyperparameters, then we will use train+val to get final mIoU in testing set

**Comment by NanqingD** [comment]
@John1231983 Thanks for your suggestions. I will try to follow up.

**Comment by NanqingD** [comment]
@John1231983 Do you mind sharing the train.txt and val.txt with me? I can redo the training.

**Comment by John1231983** [comment]
Of course, the train.txt and val.txt is 
https://gist.github.com/John1231983/0ddef9f67b9be8987486171d8e6debce
https://gist.github.com/John1231983/8b1192343ce3e779d6a5cfa7e9301f59

The number of training is 10582 images and validation is 1449 as the paper mention. Note that, this is SegmentationClassAug in the https://www.dropbox.com/s/oeu149j8qtbs1x0/SegmentationClassAug.zip?dl=0

Good job.

**Comment by John1231983** [comment]
Hi again, I am running your code in my server, It looks converge too slow. Could you share the snapshot that you completed? I want to see the performance in validation set. This is my current log

```
step 257779, tot_loss = 2.106363, seg_loss = 1.371428, reg_loss = 0.734934, mean_iou: 0.296217, lr: 0.006033(0.662 sec/step)
step 257780, tot_loss = 2.439267, seg_loss = 1.704340, reg_loss = 0.734927, mean_iou: 0.296218, lr: 0.006033(0.666 sec/step)
step 257781, tot_loss = 1.998012, seg_loss = 1.263092, reg_loss = 0.734919, mean_iou: 0.296218, lr: 0.006033(0.662 sec/step)
```
